tests/create_survey_dcip_test.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+# pylint: disable=R0914
+
+import tempfile
+from pathlib import Path
+
+import numpy as np
+
+from geoh5py.objects import CurrentElectrode, PotentialElectrode
+from geoh5py.shared.utils import compare_entities
+from geoh5py.workspace import Workspace
+
+
+def test_survey_dcip():
+
+    name = "TestCurrents"
+    n_data = 12
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        path = Path(tempdir) / r"testDC.geoh5"
+
+        # Create a workspace
+        workspace = Workspace(path)
+
+        # Create sources along line
+        x_loc, y_loc = np.meshgrid(np.arange(n_data), np.arange(-1, 3))
+        vertices = np.c_[x_loc.ravel(), y_loc.ravel(), np.zeros_like(x_loc).ravel()]
+        parts = np.kron(np.arange(4), np.ones(n_data)).astype("int")
```

Comment:
Isn't this the same as np.tile(np.arange(4), n_data)?? I don't think too many people outside of Eldad's circle know about the kronecker product trick, maybe best to avoid for the 1d array case?

Reply:
Did not know about tile. But no unfortunately. Tile would return [0, 1, 0, 1, 0, 1 ...], while the kron returns [0, 0, 0, ..., 1,1,1]. I'm open to a different function ... if you can think of one